### ISSUE TITLE
[package] b2 is built with too modern glibc on a bintray or missing classifier to distinguish

### ISSUE THREAD

**Issue by fulara** [conan-io/conan-center-index#1116]
We wanted to use b2 from conan-bintray.
To our surprise though - it did not require build on our rh6, so when we tried to use it.. it does not work.

This totally breaks our usage of boost from conan-bintray we are unable to use it. 
Historically it seems to have been baked which has into boost package now boost conanfile fetches it as a dependency.

So the problem in summary is:
```
/var/build/fulara/conan-recipes/.conan-standalone/bamboo/.conan/data/b2/4.1.0/_/_/package/44fcf6b9a7fb86b2586303e3db40189d3b511830/bin/b2
/var/build/fulara/conan-recipes/.conan-standalone/bamboo/.conan/data/b2/4.1.0/_/_/package/44fcf6b9a7fb86b2586303e3db40189d3b511830/bin/b2: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.17' not found (required by /var/build/fulara/conan-recipes/.conan-standalone/bamboo/.conan/data/b2/4.1.0/_/_/package/44fcf6b9a7fb86b2586303e3db40189d3b511830/bin/b2)
/var/build/fulara/conan-recipes/.conan-standalone/bamboo/.conan/data/b2/4.1.0/_/_/package/44fcf6b9a7fb86b2586303e3db40189d3b511830/bin/b2: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by /var/build/fulara/conan-recipes/.conan-standalone/bamboo/.conan/data/b2/4.1.0/_/_/package/44fcf6b9a7fb86b2586303e3db40189d3b511830/bin/b2)
/var/build/fulara/conan-recipes/.conan-standalone/bamboo/.conan/data/b2/4.1.0/_/_/package/44fcf6b9a7fb86b2586303e3db40189d3b511830/bin/b2: /lib64/libc.so.6: version `GLIBC_2.17' not found (required by /var/build/fulara/conan-recipes/.conan-standalone/bamboo/.conan/data/b2/4.1.0/_/_/package/44fcf6b9a7fb86b2586303e3db40189d3b511830/bin/b2)
```
glibc available on rh6 is too old for the b2 build by you guys. 
It works on rh7.

I think its missing some kind of setting or something like that, however this is a blocker for us.
As a solution we will use --build b2 and upload our own version of b2 into our artifactory. 

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **boost/1.71.0**
  * Operating System+version: **Redhat 6.10 santiago**
  * Compiler+version: **GCC 8**
  * Conan version: **conan 1.23.0**
  * Python version: **Python 3.6.10**

**Comment by prince-chrismc** [comment]
Are you using [devtoolset-8](https://access.redhat.com/documentation/en-us/red_hat_developer_toolset/8/html/8.0_release_notes/dts8.0_release)?

**Comment by fulara** [comment]
hello @prince-chrismc 
yes, I am using devtoolset-8 however that does not make any difference. the one i have `--build b2` was built with devtoolset-8, but probably dt-3 would have worked as well.

the b2 binary is not runnable on rh6.  
and devtoolset-8 does not provide any ld_library_path or anything when `source enable` I even tried adding all the dirs into `LD_LIBRARY_PATH` but that did not help

**Comment by prince-chrismc** [comment]
Hmm that is very interesting. That has not been my experience with the devtoolset however I have only ran on Centos7 which may be packaged differently. 

Typically there is always a `LD_LIBRARY_PATH` which provides the correct GLIBC when within `source enable` context.

**Comment by fulara** [comment]
Below information is based solely on my experience  and from what I heard, i have no read documentation.

My understanding on how devtoolsets are supposed to work is that they have built in diff between system glibc and glibc they want to use, that diff is then statically linked into applications.
devtoolset do indeed provide some `LD_LIBRARY_PATH` but I dont know the extents of that.

Currently we have added a custom settings to our settings.yml called 'os.version' which corresponds to major release 6 or 7. 
Thanks to this settings packages built on 6 do not mix with packages built on 7, they require rebuild.

We heavily depends on functionality where our target boxes do not have devtoolsets installed, but only the build machines have it.

It is my experience that binaries build on rh6 do run on rh7 but not the other way around.

**Comment by fulara** [comment]
I am closing this one.
I learned that I was too naitve in my aproach.
I should have expanded os_build  for my systems. we have only done so for 'os'. 
None of the recipes from conan-center-index works on centos6.

I think the choice taken by conan is too naive, it automatically matches too many distributions.

For anyone maybe reading this in the future:
IF you are thinking of using conan on rh6 or rh7 whatever, expand your profile by:
os_build -> Add a child classifier there for major OS build.
os -> Add a child classifier there for major OS build.
Thats what I learned.

Unfortunatelly installing ANY package out of conan-center-index with binaries will lead to problems.
